### PR TITLE
feat: add xsuaa provider

### PR DESCRIPTION
**Pull request by zanfee**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The [XSUAA service](https://learning.sap.com/learning-journey/discover-sap-business-technology-platform/illustrating-sap-authorization-and-trust-management-service-xsuaa-_b9fde282-4cff-4dca-b146-7c8f8dde9955) handles authorization for Cloud Foundry projects within SAP BTP. We have successfully deployed a proof of concept on Cloud Foundry using this provider. However, I was not able to launch the playground to test the integration inside this module.

Nuxt is rarely used in the SAP community as they have their own frameworks which are based on OData and come with UI components included. We are also just starting to experiment with it. I hope to lower the barrier to use Nuxt in the SAP ecosystem and open it up to more use cases.

### 📝 Checklist
- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Atinux** [comment]
Thank you @zanfee !

Do you know how I can make it work for the playground to test?

**Comment by zanfee** [comment]
I guess you'd need access to the xsuaa service. Not sure if you need to have access to sap btp or if you can access it via cloud foundry directly. At Bechtle we have access so I can try it out if i'm able to start the playground once i'm back from vacation in two weeks.

File: playground/.env.example [hunk 1]
```diff
@@ -49,3 +49,6 @@ NUXT_OAUTH_STEAM_API_KEY=
 # X
 NUXT_OAUTH_X_CLIENT_ID=
 NUXT_OAUTH_X_CLIENT_SECRET=
+# XSUAA
+NUXT_OAUTH_XSUAA_CLIENT_ID=
+NUXT_OAUTH_XSUAA_CLIENT_SECRET=
```
Comment (Atinux) [inline playground/.env.example:54]:
```suggestion
NUXT_OAUTH_XSUAA_CLIENT_SECRET=
NUXT_OAUTH_XSUAA_DOMAIN=
```

**Comment by Atinux** [comment]
Well merging for now, tell me if it works, if not you can make another PR later to fix :)

**Comment by zanfee** [comment]
Thanks, I'll give it try. I think this version isn't published on npm yet?

**Comment by zanfee** [comment]
I successfully tested login, logout and querying basic user data in my app :)

**Pull request merged**